test/server/routes/oauth.test.ts

```diff
@@ -342,5 +357,68 @@ describe('server/routes/oauth', () => {
         error_description: 'Invalid request: `redirect_uri` is invalid',
       });
     });
+
+    describe('PKCE', async () => {
+      let authorization, validParams, codeVerifier, codeChallenge;
+
+      beforeEach(async () => {
+        codeVerifier = generatePKCECodeVerifier();
+        codeChallenge = await calculatePKCECodeChallenge(codeVerifier);
+
+        authorization = await fakeOAuthAuthorizationCode({
+          codeChallenge: codeChallenge,
+          codeChallengeMethod: 'S256',
+        });
+
+        validParams = {
+          grant_type: 'authorization_code',
+          code: authorization.code,
+          client_id: authorization.application.clientId,
+          client_secret: authorization.application.clientSecret,
+          redirect_uri: authorization.application.callbackUrl,
+          code_verifier: codeVerifier,
+        };
+      });
+
+      it('succeeds with a valid code_verifier', async () => {
+        await request(expressApp)
+          .post('/oauth/token')
+          .type(`application/x-www-form-urlencoded`)
+          .send({ ...validParams })
+          .expect(200);
+      });
+
+      it('must provide a code_verifier', async () => {
+        const response = await request(expressApp)
+          .post('/oauth/token')
+          .type(`application/x-www-form-urlencoded`)
+          .send({
+            grant_type: 'authorization_code',
+            code: authorization.code,
+            client_id: authorization.application.clientId,
+            client_secret: authorization.application.clientSecret,
+            redirect_uri: authorization.application.callbackUrl,
+          })
```

Comment:
I had been using `code_verifier: undefined` but it turns out superagent doesn't strip undefined values when sending json, so errors happened here. So instead I'm just taking the `validParams` from above and inlining them without the `code_verifier` parameter.

This parameter is only required because the Authorization Code was created with PKCE.